src/main/java/io/quarkus/security/identity/RunAsUser.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.security.identity;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.security.Principal;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Identifies a method that should be executed with a new, temporary {@link SecurityIdentity}.
```

Comment:
```suggestion
 * Identifies a method that must be run with a new {@link SecurityIdentity} that is valid only for the duration of this method's execution.
```

Proposing to be more specific about `temporary`, `should be executed` is fine, but I replaced it with `must be run` to avoid some repetition with the proposed `this method's execution` at the end